python/setup.py

```diff
@@ -104,7 +105,7 @@
 
 def _supports_symlinks():
     """Check if the system supports symlinks (e.g. *nix) or not."""
-    return getattr(os, "symlink", None) is not None
+    return getattr(os, "symlink", None) is not None and ctypes.windll.shell32.IsUserAnAdmin() != 0 if sys.platform == "win32" else True
```

Comment:
You will have to do something like this to minimize the potential breakage:

```suggestion
    return hasattr(os, "symlink") and (
        (not hasattr(ctypes, "windll"))  # Non-Windows
        or (
            # In some Windows, `os.symlink` works only for admins.
            hasattr(ctypes, "windll")
            and hasattr(ctypes.windll, "shell32")
            and hasattr(ctypes.windll.shell32, "IsUserAnAdmin")
            and ctypes.windll.shell32.IsUserAnAdmin() != 0
        )
    )
```